ocaml/xapi/xapi_pgpu_helpers.ml

```diff
@@ -109,14 +109,17 @@ let get_remaining_capacity_internal ~__context ~self ~vgpu_type =
        * VM, the remaining capacity is binary. We simply return 0 if the PCI
        * device is currently passed-through to a VM, or is scheduled to be,
        * and 1 otherwise. *)
+      let inuse = List.exists (fun (_,pgpu) -> pgpu = self ) pre_allocate_list in
       let pci = Db.PGPU.get_PCI ~__context ~self in
       let scheduled = List.length (get_scheduled_VGPUs ~__context ~self) > 0 in
       let attached = Db.PCI.get_attached_VMs ~__context ~self:pci <> [] in
-      convert_capacity (if scheduled || attached then 0L else 1L)
+      convert_capacity (if scheduled || attached || inuse then 0L else 1L)
     else begin
       (* For virtual VGPUs, we calculate the number of times the VGPU_type's
        * size fits into the PGPU's (size - utilisation). *)
       let pgpu_size = Db.PGPU.get_size ~__context ~self in
+      let pgpu_pre_allocatd_vGPUs = List.filter (fun (vgpu,pgpu)->pgpu=self) pre_allocate_list
```

Comment:
Typo `pgpu_pre_allocatd_vGPUs` -> `pgpu_pre_allocated_vGPUs`. Also please add some spacing around the operators `->` and `=` (and `@` below), and use `_` for unused parameters.